lib/debug/repl.cpp

```diff
@@ -28,29 +27,30 @@ void Repl::addHelpCommand()
 
 void Repl::addAdcCommands(const uint8_t pin)
 {
-    const auto adc = std::make_shared<hyped::io::Adc>(pin, log_);
-    {
-      Command command;
-      std::stringstream identifier;
-      identifier << "adc" << static_cast<int>(pin) << " read";
-      command.name = identifier.str();
-      std::stringstream description;
-      description << "Read from ADC pin " << static_cast<int>(pin);
-      command.description = description.str();
-      command.callback = [this, adc]() {
-        const auto value = adc->readValue();
-        if (value) {
-          log_.log(hyped::core::LogLevel::kInfo, "ADC value: %d", *value);
-        } 
-      };
-    }
+  const auto adc = std::make_shared<hyped::io::Adc>(pin, log_);
+  {
+    Command command;
+    std::stringstream identifier;
+    identifier << "adc" << static_cast<int>(pin) << " read";
+    command.name = identifier.str();
+    std::stringstream description;
+    description << "Read from ADC pin " << static_cast<int>(pin);
+    command.description = description.str();
+    command.handler     = [this, adc]() {
+      const auto value = adc->readValue();
+      if (value) { log_.log(hyped::core::LogLevel::kInfo, "ADC value: %d", *value); }
+    };
+  }
 }
 
 void Repl::printCommands()
 {
   log_.log(hyped::core::LogLevel::kInfo, "Available commands:");
-  for (const auto &cmd : commands_) {
-    log_.log(hyped::core::LogLevel::kInfo, "  %s: %s", cmd.name.c_str(), cmd.description.c_str());
+  for (const auto &command : command_map_) {
```

Comment:
this isn't just a `command` either

Reply:
actually, you might want to use https://en.cppreference.com/w/cpp/language/structured_binding here and below

Reply:
You're right, you can't!